test/lib/apus/adapters/twilio_adapter_test.exs

```diff
@@ -12,15 +12,30 @@ defmodule Apus.TwilioAdapterTest do
         auth_token: "fake-auth-token"
       }
 
-      message = Message.new(from: "+15551234567", to: "+15557654321", body: "Hello there")
+      message =
+        Message.new(
+          from: "+15551234567",
+          to: "+15557654321",
+          body: "Hello there",
+          provider: nil,
+          message_id: nil,
+          status_callback: "https://valid_url.com"
+        )
 
       use_cassette "twilio_sms_from_success", match_requests_on: [:request_body] do
         {:ok, %{} = tw_message} = TwilioAdapter.deliver(message, config)
 
         assert tw_message.from == "+15551234567"
         assert tw_message.to == "+15557654321"
         assert tw_message.body == "Hello there"
+        assert tw_message.message_id == "SM123"
+        assert tw_message.provider == "twilio"
       end
+
+      cassette = File.read!("fixture/vcr_cassettes/twilio_sms_from_success.json")
+
+      assert cassette =~
+               "\"body\": \"Body=Hello+there&From=%2B15551234567&StatusCallback=https%3A%2F%2Fvalid_url.com&To=%2B15557654321\""
```

Comment:
What's the purpose of asserting the mock file?

Reply:
purely to ensure the statuscallback gets sent back

Reply:
I'm not sure I fully understand. The status callback will get asserted in the request by the `match_requests_on: [:request_body]` as part of the `use_cassette` macro above. If that is missing in either the cassette file or in the adapter, this test should be failing.

I don't think we need to have assertions for the static mock files, as changing the data in these files will cause tests to fail if the removed/changed data was important.